src/Plugin/QueueWorker/AbstractPostProcessorQueueWorker.php

```diff
@@ -0,0 +1,631 @@
+      $plugin_instance = $this->strawberryRunnerProcessorPluginManager->createInstance(
+        $plugin_config_entity->getPluginid(),
+        $configuration_options
+      );
+      return $plugin_instance;
+    }
+  }
+
+
+  /**
+   * Gets all Children of the currently being processed Processor Plugin
+   *
+   * @param string $current_id
+   *
+   * @return array
+   */
+  private function getChildProcessorIds(string $plugin_config_entity_id):array {
+    /* @var $plugin_config_entities \Drupal\strawberry_runners\Entity\strawberryRunnerPostprocessorEntity[] */
+    $plugin_config_entities = $this->entityTypeManager->getListBuilder('strawberry_runners_postprocessor')
+      ->load();
+    $active_plugins = [];
+    // This kids should be cached;
+    // We basically want here what type of processor this is and its input_argument and input_options
+    $plugin_definitions = $this->strawberryRunnerProcessorPluginManager->getDefinitions();
+
+    error_log('getting child processors');
+    foreach ($plugin_config_entities as $plugin_config_entity) {
+      // Only get first level (no Parents) and Active ones.
+      if ($plugin_config_entity->isActive() && $plugin_config_entity->getParent() == $plugin_config_entity_id) {
+        $active_plugins[] = [
+          'config_entity' => $plugin_config_entity,
+          'plugin_definition' => $plugin_definitions[$plugin_config_entity->getPluginid()]
+          ];
+      }
+    }
+    return $active_plugins;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($data) {
+
+    $processor_instance = $this->getProcessorPlugin($data->plugin_config_entity_id);
+
+    $processor_config = $processor_instance->getConfiguration();
+
+    if (!isset($data->fid) || $data->fid == NULL || !isset($data->nid) || $data->nid == NULL || !is_array($data->metadata)) {
+      return;
+    }
+    $file = $this->entityTypeManager->getStorage('file')->load($data->fid);
+    // 0 byte files have checksum, check what it is!
+    if ($file === NULL || !isset($data->metadata['checksum'])) {
+      error_log('Sorry the file does not exist or has no checksum yet. We really need the checksum');
+      return;
+    }
+
+    //We only deal with NODES.
+    $entity = $this->entityTypeManager->getStorage('node')
+      ->load($data->nid);
+
+    if (!$entity) {
+      error_log('Sorry the Node ID passed to this processor does not exist.');
+    }
+
+    //@TODO should we wrap this around a try catch?
+    $filelocation = $this->ensureFileAvailability($file);
+
+    if ($filelocation === NULL) {
+      return;
+    }
+    // Means we could pass also a file directly anytime
+    $data->filelocation = $filelocation;
+    $tobeindexed = FALSE;
+    error_log(in_array('searchapi',$processor_config['output_destination']));
+    // make all this options constants
+    if (array_key_exists('searchapi', $processor_config['output_destination']) && $processor_config['output_destination']['searchapi'] == 'searchapi') {
```

Comment:
@DiegoPino I found why this doesn't work, its a PHP issue:
Here https://wiki.php.net/rfc/string_to_number_comparison you can read that:
_Comparisons between strings and numbers using == and other non-strict comparison operators currently work by casting the string to a number, and subsequently performing a comparison on integers or floats. This results in many surprising comparison results, the most notable of which is that 0 == "foobar" returns true._ 
And that is just really our case where array value is "0" and compared with string return TRUE!!!
We can simply solve it adding a triple ===, and the it works!

Reply:
@DiegoPino Sorry, I really don't want force a change request. it was my fault using git ... It was intended as a simple comment, sorry.

Reply:
@giancarlobi this is perfect. Change request is the correct way and grateful you did this. So silly of me, makes total sense to not compare via a == is should have done it via === . Good catch I was totally aware but also totally blind to this! hahaha. Will fix ASAP. You good!

Reply:
If this works, please feel free to approve and we can merge and then do more work on other issues.thx!

Reply:
I already approved as reviewer, is it right?